main.py

```diff
@@ -96,39 +94,39 @@ def _train(pipeline_name, dev_mode):
 @action.command()
 @click.option('-p', '--pipeline_name', help='pipeline to be trained', required=True)
 @click.option('-d', '--dev_mode', help='if true only a small sample of data will be used', is_flag=True, required=False)
-def evaluate(pipeline_name, dev_mode):
+@click.option('-c', '--chunk_size', help='size of the chunks to run evaluation on', type=int, default=None,
+              required=False)
+def evaluate(pipeline_name, dev_mode, chunk_size):
     logger.info('evaluating')
-    _evaluate(pipeline_name, dev_mode)
+    _evaluate(pipeline_name, dev_mode, chunk_size)
 
 
-def _evaluate(pipeline_name, dev_mode):
+def _evaluate(pipeline_name, dev_mode, chunk_size):
     meta = pd.read_csv(os.path.join(params.meta_dir, 'stage{}_metadata.csv'.format(params.competition_stage)))
     meta_valid = meta[meta['is_valid'] == 1]
 
     if dev_mode:
         meta_valid = meta_valid.sample(30, random_state=1234)
 
-    data = {'input': {'meta': meta_valid,
-                      'meta_valid': None,
-                      'train_mode': False,
-                      'target_sizes': [(300, 300)] * len(meta_valid),
-                      },
-            }
-
     pipeline = PIPELINES[pipeline_name]['inference'](SOLUTION_CONFIG)
-    output = pipeline.transform(data)
-    pipeline.clean_cache()
-    y_pred = output['y_pred']
-    pipeline.clean_cache()
+    prediction = generate_prediction(meta_valid, pipeline, logger, CATEGORY_IDS, chunk_size)
+
+    prediction_filepath = os.path.join(params.experiment_dir, 'prediction.json')
+    with open(prediction_filepath, "w") as fp:
+        fp.write(json.dumps(prediction))
 
-    y_true = read_masks(meta_valid[Y_COLUMNS_SCORING].values, params.data_dir, dataset="val")
+    annotation_file_name = "annotation.json"
+    annotation_file_path = os.path.join(params.data_dir, 'val', annotation_file_name)
```

Comment:
@taraspiotr annotation_file_name variable seems like an overkill how about we just pass it to os.path.join ?

Reply:
@jakubczakon I agree, changed.